packages/vue-components/src/Announcement.vue

```diff
@@ -0,0 +1,147 @@
+<!-- Announcement.vue -->
+<template>
+  <div
+    v-if="!dismissed"
+    ref="announcement"
+    :class="['announcement', 'text-center', themeClass]"
+    :style="positionStyle"
+  >
+    <div class="announcement-content">
+      <slot></slot>
+    </div>
+    <button
+      v-if="dismissible"
+      type="button"
+      class="btn-close"
+      aria-label="Close"
+      @click="dismiss"
+    ></button>
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    dismissible: {
+      type: Boolean,
+      default: false,
+    },
+    theme: {
+      type: String,
+      default: 'warning',
+      validator: value => [
+        'primary', 'secondary', 'success', 'warning',
+        'danger', 'info', 'light', 'dark',
+      ].includes(value),
+    },
+    placement: {
+      type: String,
+      default: 'top',
+      validator: value => [
+        'top', 'top-right', 'top-left', 'bottom',
+        'bottom-right', 'bottom-left',
+      ].includes(value),
+    },
+  },
+  data() {
+    return {
+      dismissed: false,
+      positionStyle: {},
+    };
+  },
+  computed: {
+    themeClass() {
+      return `bg-${this.theme}`;
+    },
+  },
+  mounted() {
+    this.setPosition();
+    window.addEventListener('resize', this.setPosition);
+  },
+  beforeUnmount() {
+    window.removeEventListener('resize', this.setPosition);
+  },
+  methods: {
+    dismiss() {
+      this.dismissed = true;
+    },
+    setPosition() {
```

Comment:
Since there are some repeating codes (width and height), would it be better to refactor it in this way:
```
// create a map
const placementStyles = {
  "top": { 
    top: '20px', 
    left: '50%', 
    transform: 'translateX(-50%)' 
  },
  "top-right": { 
    top: '20px', 
    right: '20px' 
  },
  "top-left": { 
    top: '20px', 
    left: '20px' 
  },
  "bottom": { 
    bottom: '20px', 
    left: '50%', 
    transform: 'translateX(-50%)' 
  },
  "bottom-right": { 
    bottom: '20px', 
    right: '20px' 
  },
  "bottom-left": { 
    bottom: '20px', 
    left: '20px' 
  }
}

setPosition() {
  const announcementElement = this.$refs.announcement;
  if (!announcementElement) return;
  const { width, height } = announcementElement.getBoundingClientRect();

  // get the respective styles from the map
  const positionStyle = { ...placementStyles[this.placement] };

  // since only width and height are specified in these cases
  if (this.placement.endsWith('right') || this.placement.endsWith('left')) {
    positionStyle.width = `${width}px`;
    positionStyle.height = `${height}px`;
  }

  this.positionStyle = positionStyle;
}
```
By doing this, next time if we introduce a new placement (eg. middle-left), we can just add the corresponding style to the map.